x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -889,4 +891,46 @@ export function trainedModelsRoutes(
         }
       )
     );
+
+  /**
+   * @apiGroup TrainedModels
+   *
+   * @api {post} /internal/ml/inference_models/create_inference_endpoint/:taskType/:inferenceId Create Inference Endpoint
+   * @apiName CreateInferenceEndpoint
+   * @apiDescription Create Inference Endpoint
+   */
+  router.versioned
+    .post({
+      path: `${ML_INTERNAL_BASE_PATH}/inference_models/create_inference_endpoint/{taskType}/{inferenceId}`,
+      access: 'internal',
+      options: {
+        tags: ['access:ml:canCreateInferenceEndpoint'],
```

Comment:
Tags used in the routes need to match capabilities.
Probably the easiest way to do this is copy `canCreateTrainedModels` [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/common/types/capabilities.ts)

Reply:
@saarikabhasi I think you need to update some tests as well

Reply:
updated tests in 1c088f57dc826c2f30a8478171b1fd84539c08ed and 94f25a6f41950c41e3a6e6af0d1b1bbe53594cf2